### ISSUE TITLE
Username and password issue over reverse proxy

### ISSUE THREAD

**Issue by mascode** [evilhero/mylar#1592]
So I have Mylar sitting behind a reverse proxy with SSL. I also have the HTTP username and password set to prevent others getting in since it is facing the public. The issue that I'm having is that whenever I go to https://www.mydomain.com/mylar it repeatedly keep asking for the username and password even after I have signed in. Navigating to any page or even scrolling down on a page prompts me to enter my username and password again. 

I've also noticed that this isn't a problem if I access Mylar from the http://mylocalip:port/mylar where I'm prompted for the username and password only once. 

Unfortunately, I don't see anything in the log file that points to this particular error. 

Any ideas on what I could be doing wrong here?

**Comment by evilhero** [comment]
I honestly don't run a reverse proxy on anything, and I'm really lacking in the knowledge about it (I try, but really it's just sad).

One thing that comes to mind, what do you have within Mylar for your host configuration ? I'm wondering if you have to set the host to be the domain you're using and then set the web_root (within the confi) to the /mylar - like I said, I have no idea really just a guess and I'm sure there are others out there that would know more about this than I. Maybe someone else can chime in ...

**Comment by mascode** [comment]
I left the host as my local IP but I did try setting it to my domain in the config.ini and here's the output.

```
16-Mar-2017 14:45:56 - INFO :: MainThread : [DIRECTORY-CHECK] Found comic directory: D:\Software\Mylar
16-Mar-2017 14:45:56 - INFO :: MAIN : Successfully discovered local IP and locking it in as : 192.168.1.155
16-Mar-2017 14:45:56 - INFO :: MAIN : Branch detected & set to : master
16-Mar-2017 14:45:56 - INFO :: MAIN : [DB Module] Loading : sqlite3 as the database module to use.
16-Mar-2017 14:45:56 - INFO :: MAIN : [Cache Check] Cache directory currently set to : D:\Software\Mylar\cache/
16-Mar-2017 14:45:56 - INFO :: MAIN : Checking to see if the database has all tables....
16-Mar-2017 14:45:56 - INFO :: MAIN : Populating Base Exception listings into Mylar....
16-Mar-2017 14:45:56 - INFO :: MAIN : Populating Custom Exception listings into Mylar....
16-Mar-2017 14:45:56 - INFO :: MAIN : Ensuring DB integrity - Removing all Erroneous Comics (ie. named None)
16-Mar-2017 14:45:56 - INFO :: MAIN : Correcting Null entries that make the main page break on startup.
16-Mar-2017 14:45:56 - INFO :: MAIN : Retrieving latest version information from github
16-Mar-2017 14:46:04 - INFO :: MAIN : Mylar is up to date
16-Mar-2017 14:46:04 - INFO :: MAIN : Remapping the sorting to allow for new additions.
16-Mar-2017 14:46:04 - INFO :: MAIN : Sucessfully ordered 134 series in your watchlist.
16-Mar-2017 14:46:04 - INFO :: MAIN : Starting Mylar on http://www.mydomain.com:8090/
16-Mar-2017 14:46:06 - ERROR :: HTTPServer Thread-1 : Uncaught exception: Traceback (most recent call last):
  File "D:\Software\Mylar\mylar\logger.py", line 158, in new_run
    old_run(*args, **kwargs)
  File "C:\Python27\lib\threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "D:\Software\Mylar\lib\cherrypy\process\servers.py", line 187, in _start_http_thread
    self.httpserver.start()
  File "D:\Software\Mylar\lib\cherrypy\wsgiserver\__init__.py", line 1753, in start
    raise socket.error(msg)
error: No socket could be created

Traceback (most recent call last):
  File "D:\Software\Mylar\mylar\logger.py", line 158, in new_run
    old_run(*args, **kwargs)
  File "C:\Python27\lib\threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "D:\Software\Mylar\lib\cherrypy\process\servers.py", line 187, in _start_http_thread
    self.httpserver.start()
  File "D:\Software\Mylar\lib\cherrypy\wsgiserver\__init__.py", line 1753, in start
    raise socket.error(msg)
socket.error: No socket could be created
Failed to start on port: 8090. Is something else running?
```

So it looks like when I set the host to my domain name it does grab my local IP address okay and does register the domain name as well but the web server isn't starting because it thinks something else is running on the 8090 port. I did check just in case to see if anything was running on that port and there was nothing. Once I set the host back to my local ip it started up again without any problems.

**Comment by BRTPOB** [comment]
Which proxy are you running? I've got the exact same setup running flawlessly on Ubuntu with nginx.

**Comment by mascode** [comment]
I'm using CaddyServer for the proxy. I did check the access logs and error logs on there and there doesn't seem to be anything in there indicating as to why this is happening. After some more poking around I hopped on different browsers and found out that Chrome and Firefox display this behavior but Edge doesn't. 

As such I don't think this has anything to do with Mylar but rather how FF/Chrome handles HTTP username/passwords or perhaps something in my configuration with my proxy and webserver. As a result I'll be closing this issue.

That being said thank you both for your input, it's helped me narrow down the issue some. I'll probably end up moving everything over to a linux machine and going with the Nginx setup like @BRTPOB has.
